src/main/java/ai/knowly/langtorch/store/vectordb/integration/pinecone/PineconeVectorStore.java

```diff
@@ -35,20 +34,30 @@ public PineconeVectorStore(
     this.pineconeVectorStoreSpec = pineconeVectorStoreSpec;
   }
 
-  /** Adds the specified documents to the Pinecone vector store database. */
-  public UpsertResponse addDocuments(List<DomainDocument> documents) {
+  /** Adds the specified documents to the Pinecone vector store database.
+   *
+   * @return true if documents added successfully, otherwise false
+   */
+  @Override
+  public boolean addDocuments(List<DomainDocument> documents) {
+    if (documents.isEmpty()) return true;
+
     return addVectors(
         documents.stream()
             .map(this::createVector)
             .collect(Collectors.toCollection(ArrayList::new)));
   }
 
-  /** Adds a list of vectors to the Pinecone vector store database. */
-  private UpsertResponse addVectors(List<Vector> vectors) {
+  /** Adds a list of vectors to the Pinecone vector store database.
+   *
+   * @return true if vectors added successfully, otherwise false
+   */
+  private boolean addVectors(List<Vector> vectors) {
```

Comment:
addVectors and createVector are very similar and hard to tell what's the difference. Rename  addVectors to addToVectorStore?